### ISSUE TITLE
process.env.NODE_ENV is undefined in 1.0.18

### ISSUE THREAD

**Issue by rosszurowski** [oven-sh/bun#7663]
### What version of Bun is running?

1.0.18+36c316a24

### What platform is your computer?

Darwin 23.2.0 arm64 arm

### What steps can reproduce the bug?

Create an `index.ts` with the following contents:

```ts
console.log(process.env.NODE_ENV)
```

Run it against Bun 1.0.17 and Bun 1.0.18.

### What is the expected behavior?

The script should output `development`:

```shell
$ ./bun-1.0.17 index.ts
development
```

### What do you see instead?

It outputs `undefined`:

```shell
$ ./bun-1.0.17 index.ts
development
$ ./bun-1.0.18 index.ts
undefined
```

### Additional information

This feels pretty critical. Lots of web apps have checks for `process.env.NODE_ENV === "production"` and failing to meet those checks will likely cause lots of issues in apps running on Bun. For example:

```tsx
if (process.env.NODE_ENV === "production") {
  sendEmail()
} else {
  logToConsole() // this gets triggered with this bug
}
```

**Comment by rosszurowski** [comment]
Looks like https://github.com/oven-sh/bun/pull/7614 is a likely candidate that might've caused this.

**Comment by Jarred-Sumner** [comment]
Oh yeah, this was mistakenly a breaking change.

We were previously setting it to “development” by default, and now we are not setting it. 

We will fix this in v1.0.19

**Comment by paperclover** [comment]
the check `process.env.NODE_ENV === "production"` is not affected by this as the default was "development", mistakenly changed to `undefined`. this does affect code explicitly checking for `process.env.NODE_ENV === "development"`
